sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1116,6 +1119,298 @@ def max_deflection(self):
+           are installed on the system.
+
+        Examples
+        ========
+        There is a beam of length 8 meters. A constant distributed load of 10 KN/m
+        is applied from half of the beam till the end. There are two simple supports
+        below the beam, one at the starting point and another at the ending point
+        of the beam. A pointload of magnitude 5 KN is also applied from top of the
+        beam, at a distance of 4 meters from the starting point.
+        Take E = 200 GPa and I = 400*(10**-6) meter**4.
+
+        Using the sign convention of downwards forces being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols
+        >>> R1, R2 = symbols('R1, R2')
+        >>> b = Beam(8, 200*(10**9), 400*(10**-6))
+        >>> b.apply_load(5000, 2, -1)
+        >>> b.apply_load(R1, 0, -1)
+        >>> b.apply_load(R2, 8, -1)
+        >>> b.apply_load(10000, 4, 0, end=8)
+        >>> b.bc_deflection = [(0, 0), (8, 0)]
+        >>> b.solve_for_reaction_loads(R1, R2)
+        >>> b.plot_loading_results()
+        Figure(640x480)
+        """
+        from sympy import lambdify
+        matplotlib = import_module('matplotlib', __import__kwargs={'fromlist':['pyplot']})
+        plt = matplotlib.pyplot
+        numpy = import_module('numpy', __import__kwargs={'fromlist':['linspace']})
+        linspace = numpy.linspace
+
+        length = self.length
+        variable = self.variable
+        if subs is None:
+            subs = {}
+        for sym in self.deflection().atoms(Symbol):
+            if sym == self.variable:
+                continue
+            if sym not in subs:
+                raise ValueError('Value of %s was not passed.' %sym)
+        if self.length in subs:
+            length = subs[self.length]
+        else:
+            length = self.length
+
+        # As we are using matplotlib directly in this method, we need to change SymPy methods
+        # to numpy functions.
+        shear = lambdify(variable, self.shear_force().subs(subs).rewrite(Piecewise), 'numpy')
+        moment = lambdify(variable, self.bending_moment().subs(subs).rewrite(Piecewise), 'numpy')
+        slope = lambdify(variable, self.slope().subs(subs).rewrite(Piecewise), 'numpy')
+        deflection = lambdify(variable, self.deflection().subs(subs).rewrite(Piecewise), 'numpy')
+
+        points = linspace(0, float(length))
+
+        # Creating a grid for subplots with 2 rows and 2 columns
+        fig, axs = plt.subplots(4, 1)
+        # axs is a 2D-numpy array containing axes
+        axs[0].plot(points, shear(points), color='g')
+        axs[0].set_title("Shear Force")
+        axs[1].plot(points, moment(points), color='b')
+        axs[1].set_title("Bending Moment")
+        axs[2].plot(points, slope(points), color='m')
+        axs[2].set_title("Slope")
+        axs[3].plot(points, deflection(points), color='r')
```

Comment:
I recommend cycling through matplotlib's standard color cycler instead of explicitly naming the colors.